File: Assets/Scripts/OSC/OSCPeakLED.js
public var channel : int = 1;
public var restColor : Color = Color.black;
public var activeColor : Color = Color.red;
public var peakLEDs : PeakLED[];

class PeakLED {
	public var obj : GameObject;
	public var peakLevel : float;
	@System.NonSerialized public var mat : Material;
}

function Start () {
	
	// initialize all peakLEDs
	for(peakLED in peakLEDs) {
		peakLED.mat = peakLED.obj.gameObject.GetComponent(Renderer).material;
		SetPeakLEDColor(peakLED, restColor);
	}
}

function SetPeakLEDColor(peakLED : PeakLED, inputColor : Color) {
	peakLED.mat.SetColor("_Color", inputColor);
	peakLED.mat.SetColor("_TintColor", inputColor);
	peakLED.mat.SetColor("_Emission", inputColor);
	//peakLED.mat.SetColor("_TintColor", inputColor);
}

function OSCAudioEnvelope (input) {
	if(input.channel == channel) {
		for(peakLED in peakLEDs) {
			if(input.envelope >= peakLED.peakLevel) {
				SetPeakLEDColor(peakLED, activeColor);
			}
			else {
				SetPeakLEDColor(peakLED, restColor);
			}
		}
	}
}

File: Assets/Scripts/OSC/OSCPeakLED.cs
using UnityEngine;
using System.Collections;

public class OSCPeakLED : MonoBehaviour 
{
	public int channel = 1;
	public Color restColor = Color.black;
	public Color activeColor = Color.red;
	public PeakLED[] peakLEDs;
	
	[System.Serializable]
	public class PeakLED {
		public GameObject obj;
		public float peakLevel;
		[System.NonSerialized]
		public Material mat;
	}
	
	void Start () {
		// initialize all peakLEDs
		foreach(PeakLED peakLED in peakLEDs) {
			peakLED.mat = peakLED.obj.gameObject.GetComponent<Renderer>().material;
			SetPeakLEDColor(peakLED, restColor);
		}
	}
	
	void SetPeakLEDColor(PeakLED peakLED, Color inputColor) {
		peakLED.mat.SetColor("_Color", inputColor);
		peakLED.mat.SetColor("_TintColor", inputColor);
		peakLED.mat.SetColor("_Emission", inputColor);
		//peakLED.mat.SetColor("_TintColor", inputColor);
	}
	
	void OSCAudioEnvelope (OSCHandler.AudioEnvelopeData input) {
		if(input.channel == channel) {
			foreach(PeakLED peakLED in peakLEDs) {
				if(input.envelope >= peakLED.peakLevel) {
					SetPeakLEDColor(peakLED, activeColor);
				}
				else {
					SetPeakLEDColor(peakLED, restColor);
				}
			}
		}
	}
}
